CMakeLists.txt

```diff
@@ -252,6 +258,37 @@ if (ENABLE_IPV6)
 endif()
 
 option(WITH_EXTERNAL_BACKEND "Support external backend provided by the application" OFF)
+if (WITH_EXTERNAL_BACKEND AND NOT WITH_LIBTINYIIOD)
+	message(SEND_ERROR "Sorry, external backend is only supported when WITH_LIBTINYIIOD=ON.")
+endif()
+
+if (WITH_EXTERNAL_BACKEND)
+	set(DISABLED_FEATURES_FOR_EXTERNAL_BACKED "")
+
+	if (WITH_UTILS)
+		list(APPEND DISABLED_FEATURES_FOR_EXTERNAL_BACKED utils)
+		set(WITH_UTILS OFF)
+	endif()
+
```

Comment:
do we need the `if (WITH_UTILS)`? and the others if's below? I was expecting just to have this do `set(WITH_UTILS OFF)` all the time.